File: lua/ht/conf/lsp/servers/pyright.lua
local CoreConst = require("ht.core.const")
local CoreLspServer = require("ht.core.lsp.server")

---@type ht.lsp.ServerOpts
local opts = {}

opts.name = "pyright"

opts.setup = function(on_attach, capabilities)
  require("lspconfig").pyright.setup {
    cmd = {
      CoreConst.mason_bin .. "/pyright-langserver",
      "--stdio",
    },
    on_attach = on_attach,
    capabilities = capabilities,
  }
end

return CoreLspServer.new(opts)

File: src/conf/external_tools/lsp_servers/pyright.ts
import { LspServer } from "@core/model";

export const server = new LspServer({
  name: "pyright",
  exe: false,
  setup: (_server: LspServer, on_attach, capabilities) => {
    luaRequire("lspconfig").pyright.setup({
      cmd: ["pyright-langserver", "--stdio"],
      on_attach,
      capabilities,
    });
  },
});
